### PR TITLE
149 csv upload

### PR DESCRIPTION
**Pull request by ezramitchell**

### CODE REVIEW

File: restaurant-api/src/main/resources/application.properties [hunk 1]
```diff
@@ -6,4 +6,6 @@ spring.datasource.password=password
 spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.PostgreSQLDialect
 spring.jpa.hibernate.ddl-auto=update
 spring.jpa.show-sql=true
-jwt.secret=secret
\ No newline at end of file
+jwt.secret=secret
+csv.bucket=qqd-temp
+csv.region=us-east-1
```
Comment (szhansmoothstack) [inline restaurant-api/src/main/resources/application.properties:11]: Why are we switching from us-east-2 to us-east1?

File: restaurant-api/src/main/java/com/smoothstack/user/service/S3Service.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+@Service
+public class S3Service {
+
+    @Value("${csv.bucket}")
+    private String bucketName;
+
+    private static final String[] expectedCsvColumns = {
+            "address",
+            "average_rating",
```
Comment (jasonwittmuss) [inline restaurant-api/src/main/java/com/smoothstack/user/service/S3Service.java]: Average Rating will not be supplied by the admin, default value will be used until ratings are given by users

File: restaurant-api/src/main/java/com/smoothstack/user/api/RestaurantApiController.java [hunk 3]
```diff
@@ -213,12 +217,23 @@ public RestaurantApiController(SearchService searchService,
...
 
+    }
 
-    (@ApiParam(value = "Non null properties will be updated, id necessary") @Valid @RequestBody(required = false) RestaurantEntity restaurant) {
-        return new ResponseEntity<>(HttpStatus.NOT_IMPLEMENTED);
+    @PostMapping(
+            path = "/restaurants/csv",
+            consumes = MediaType.MULTIPART_FORM_DATA_VALUE
+    )
+    public ResponseEntity<Void> uploadCSV(@RequestParam("File") MultipartFile file, @RequestParam(value = "FileName", defaultValue = "") String fileName) throws IllegalArgumentException{
```
Comment (szhansmoothstack) [inline restaurant-api/src/main/java/com/smoothstack/user/api/RestaurantApiController.java:230]: Probably need to secure both the uploadCSV and update methods

File: restaurant-api/src/main/resources/application.properties (hunk 1 above)
Comment (ezramitchell) [reply restaurant-api/src/main/resources/application.properties:11]: The region switch was due to a change in our base infrastructure. Cloudfront distributions require ACM certificates from us-east-1 so the base infrastructure was switched.

File: restaurant-api/src/main/java/com/smoothstack/user/api/RestaurantApiController.java (hunk 3 above)
Comment (ezramitchell) [reply restaurant-api/src/main/java/com/smoothstack/user/api/RestaurantApiController.java:230]: The method is now secured

File: restaurant-api/src/main/java/com/smoothstack/user/service/S3Service.java (hunk 2 above)
Comment (ezramitchell) [reply restaurant-api/src/main/java/com/smoothstack/user/service/S3Service.java]: average rating has been removed

**Review by szhansmoothstack — APPROVED** [review]

**Review by jasonwittmuss — APPROVED** [review]

**Pull request merged**